compiler/src/iree/compiler/Dialect/Flow/Transforms/Passes.cpp

```diff
@@ -175,120 +114,10 @@ static void addCleanupPatterns(OpPassManager &passManager) {
-      //       TODO: This pass is only needed till all backends can handle
-      //       multiple reduction dimensions.
-      .addPredicatedPass(clCollapseReductionDims,
-                         IREE::Flow::createCollapseReductionDimensionsPass)
-
-      //     c. Split reduction operations into parallel and reduction, i.e
-      //        .
-      .addPass(IREE::Flow::createSplitReductionPass)
-
-      //     d. Transpose generic ops to
-      //        - help with dispatch region formation.
-      //        - move reduction iterators to be innermost.
-      .addPass(IREE::Flow::createTransposeGenericOpsPass);
-}
-
-// Pipeline to first create `flow.dispatch.region` ops and then lower to
-// `flow.dispatch.workgroup` ops.
-static void addDispatchRegionCreationPasses(OpPassManager &passManager) {
-  FunctionLikeNest(passManager)
-      // Only want use the transform dialect for some dispatch regions and let
-      // the FormDispatchRegions handle the rest. This only moves the root
-      // compute op into the dispatch region, so that we can run additional
-      // transformations afterwards with a simple region and without bothering
-      // producers.
-      .addPredicatedPass(
-          !clDispatchTransformFileName.empty(),
-          [&]() {
-            DispatchWithTransformDialectPassOptions options;
-            options.transformSpecPath = clDispatchTransformFileName;
-            return createDispatchWithTransformDialectPass(options);
-          })
-      // Create dispatches for scalar operations as roots
-      .addPass(IREE::Flow::createFormScalarDispatchesPass)
-      // Create `flow.dispatch.region` centered around a root and fuse with
-      // producers and consumers.
-      .addPass([&]() {
-        return IREE::Flow::createFormDispatchRegionsPass(
-            FormDispatchRegionsPassOptions{
-                clEnableAggressiveFusion,
-                clEnableFusePaddingIntoLinalgConsumerOps,
-                clEnableFusePaddingIntoLinalgProducerOps});
-      })
-      // Clone all producers into the dispatch region to perpare for being
-      // isolated from above. This enables running additional transformations
-      // afterwards that would need the full dispatch content but don't want to
-      // handle explicit captures as materialized as dispatch workgroup operands
-      // and block arguments.
-      .addPass(IREE::Flow::createCloneProducersIntoDispatchRegionsPass)
-      .addPredicatedPass(clEnableDataTiling,
-                         [&]() {
-                           return createSetEncodingPass(
-                               SetEncodingPassOptions{clPadFactor});
-                         })
-      // Collapse dimensions of linalg Ops.
-      .addPass(IREE::Flow::createCollapseDimensionsPass)
-      // Convert dispatch regions into dispatch workgroups by capturing values
-      // and making the new workgroups isolated from above.
       .addPass(IREE::Flow::createConvertDispatchRegionsToWorkgroupsPass)
```

Comment:
Might be easier to move all the remaining passes also into dispatch region formation pipeline that you are adding. They are really specific to how we create dispatches by default. THat might help with the dependencies. 
You could still keep the utility to convert the `flow.dispatch.region` -> `flow.dispatch.workgroups` in Flow, but move the passes if that works.